File: Binary tree/Depth_first_Search/java/PostOrderTraversal.java
class Node{
    int data;
    Node left ;
    Node right;

    public Node(int data){
        this.data= data;
        left = null ;
        right = null;
    }

}

public class PostOrderTraversal{
    

    public void postOrder(Node root){
        if(root==null){
            return;
        }
        postOrder(root.left);
        postOrder(root.right);
        System.out.println(root.data);

    }
    

    public static void main(String args[]){

        Node root =new Node(1);
        root.left= new Node(2);
        root.right = new Node(3);
        root.left.left = new Node(4);
        root.left.right= new Node(5);
        root.right.left=new Node(6);
        root.right.right=new Node(7);

        PostOrderTraversal traversal = new PostOrderTraversal();
        traversal.postOrder(root);

    }
}

File: Binary tree/Depth_first_Search/javascript/PostOrderTraversal.js
class Node {
  constructor(data) {
    this.data = data;
    this.left = null;
    this.right = null;
  }
}

class postOrderTraversal {
  postOrder(root) {
    if (root === null) {
      return;
    }
    this.postOrder(root.left);
    this.postOrder(root.right);
    console.log(root.data);
  }
}

function main() {
  let root = new Node(1);
  root.left = new Node(2);
  root.right = new Node(3);
  root.left.left = new Node(4);
  root.left.right = new Node(5);
  root.right.left = new Node(6);
  root.right.right = new Node(7);

  let traversal = new postOrderTraversal();
  traversal.postOrder(root);
}
main();
